tests/components/smlight/test_button.py

```diff
@@ -61,17 +73,50 @@ async def test_buttons(
     mock_method.assert_called_with()
 
 
-@pytest.mark.usefixtures("mock_smlight_client")
-async def test_disabled_by_default_button(
+@pytest.mark.parametrize("entity_id", ["zigbee_flash_mode", "reconnect_zigbee_router"])
+async def test_disabled_by_default_buttons(
     hass: HomeAssistant,
+    entity_id: str,
     entity_registry: er.EntityRegistry,
     mock_config_entry: MockConfigEntry,
+    mock_smlight_client: MagicMock,
 ) -> None:
-    """Test the disabled by default flash mode button."""
+    """Test the disabled by default buttons."""
+    if entity_id == "reconnect_zigbee_router":
```

Comment:
We should not have if statements in test code

Reply:
ok, I guess for now will run all button tests as router, since it has all buttons available. Might change in the future though if other conditional buttons get added.